core/storage/couchStorageHelpers.go

```diff
@@ -0,0 +1,175 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	_ "github.com/go-kivik/couchdb/v3" // The CouchDB Driver
+	kivik "github.com/go-kivik/kivik/v3"
+	"github.com/open-horizon/edge-sync-service/common"
+	"io/ioutil"
+	"strings"
+	"time"
+)
+
+//Add better error-handling for functions (meaningful messages)
+
+func (store *CouchStorage) getOne(id string, result interface{}) common.SyncServiceError {
+
+	if exists, err := store.client.DBExists(context.TODO(), store.loginInfo["dbName"]); err != nil || !exists {
+		return err
+	}
+
+	db := store.client.DB(context.TODO(), store.loginInfo["dbName"])
+
+	row := db.Get(context.TODO(), id)
+	if err := row.ScanDoc(&result); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (store *CouchStorage) addAttachment(id string, data []byte) common.SyncServiceError {
+
+	if exists, err := store.client.DBExists(context.TODO(), store.loginInfo["dbName"]); err != nil || !exists {
+		return err
+	}
+
+	db := store.client.DB(context.TODO(), store.loginInfo["dbName"])
+
+	var rev string
+	row := db.Get(context.TODO(), id)
+	if row == nil {
+		return &Error{fmt.Sprintf("Object not found")}
```

Comment:
Oh right okay, but if I do that, the function would return common.NotFound (instead of common.SyncServiceError) right? In that case, how would other errors be detected? The function that calls this helper function, returns the err if err is anything other than 'not found'. If it is 'not found' then it simply assigns existingObject = nil

Reply:
Oh okay right, this is definitely a better way. I will make the change everywhere